frontend/src/pages/UserPage.jsx

```diff
@@ -1,13 +1,192 @@
+      const newStatus = !userData.is_active;
+      updateUserStatus(`/users/${id}`, {
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify({ is_active: newStatus }),
+      });
+    }
+  };
+
+  const closeDeactivateModal = () => {
+    setShowDeactivateModal(false);
+    if (deactivateModalRef.current) {
+      deactivateModalRef.current.close();
+    }
+  };
+
+  if (isUserLoading) {
+    return <Spinny />;
+  }
+
+  if (fetchUserError || !userData) {
+    return (
+      <div className='section-card flex flex-col items-center justify-center min-h-[60vh]'>
+        <h1 className='section-heading text-error'>Error</h1>
+        <p className='text-lg text-gray-600 mb-8'>
+          {fetchUserError?.message ||
+            'Could not load user data or user not found.'}
+        </p>
+        <Link to='/users' className='animated-button'>
+          Back to All Users
+        </Link>
+      </div>
+    );
+  }
+
+  const isUserActive = userData.is_active;
+  const modalAction = isUserActive ? 'Deactivate' : 'Activate';
+  const modalTitle = `Confirm ${isUserActive ? 'Deactivation' : 'Activation'}`;
+  const modalMessage = isUserActive
+    ? `Deactivate ${userData.name}'s account?`
+    : `Activate ${userData.name}'s account?`;
+  const confirmButtonClass = isUserActive
+    ? 'animated-button !bg-red-500 hover:!bg-red-600'
+    : 'animated-button !bg-green-500 hover:!bg-green-600';
 
   return (
-    <div className='section-card flex flex-col items-center justify-center min-h-[60vh]'>
-      <h1 className='section-heading'>User Profile: {id}</h1>
-      <p className='text-lg text-gray-600 mb-8'>
-        This page will display the profile for user with ID: {id}.
-      </p>
+    <div className='section-card flex flex-col items-center min-h-[60vh]'>
+      <h1 className='section-heading'>User Profile: {userData.name}</h1>
+      <div className='w-full max-w-xl p-4 bg-base-200 rounded-lg shadow-md'>
+        <p className='text-lg mb-2'>
+          <strong>Username:</strong> {userData.username}
+        </p>
+        <p className='text-lg mb-2'>
+          <strong>Name:</strong> {userData.name}
+        </p>
+        <p className='text-lg mb-4'>
+          <strong>Admin:</strong> {userData.is_admin ? 'Yes' : 'No'}
+        </p>
+        <p className='text-lg mb-4'>
+          <strong>Status:</strong> {isUserActive ? 'Active' : 'Inactive'}
+        </p>
+
+        <div className='flex flex-col sm:flex-row justify-center gap-4 mt-6'>
+          <Link to={`/users/${id}/edit`} className='animated-button'>
+            Edit User
+          </Link>
+          <Link to={`/users/${id}/ideas`} className='animated-button'>
+            View All Ideas
+          </Link>
+          <button
+            onClick={handleToggleStatusClick}
+            disabled={isUpdating}
+            className={confirmButtonClass}
+          >
+            {isUpdating ? 'Updating Status...' : modalAction}
```

Comment:
I imagined it'd be used this way:
```suggestion
            {isUpdating ? 'Updating Status...' : `${modalAction} Account`}
```